### DISCUSSION TITLE
Autorange based on values, ignoring shapes

### DISCUSSION THREAD

**Discussion by Qhilm** [dbuezas/lovelace-plotly-graph-card#343] [Q&A]
Hello,

I am attempting to have autorange "zoom in" the y axis based on min and max value displayed. I would add a 10% margin, like this:

min: (mininum value of entity sensor.xyz within visible time range)
max: (maxinum value of entity sensor.xyz within visible time range)
margin: (max - min)/10

range: [min - margin, max + margin]

My skills in plotly are far from sufficient to write this unfortunately. If someone can point me to a sample which is close enough, that would help me.

**Comment by dbuezas** [comment]
I am not aware of any plotly feature for that, but here's a trick you could do:
* Add the same entity again
* use a filter to multiply its value by 1.2 (+20%)
* add the same entity again but this time subtract a value. Making it symmetrical is trickier 
* make both fake entities transparent
* set `showlegend: false` for the fake entities

**Comment by Qhilm** [comment]
Thanks, the problem is ignoring the shapes I think.

How does the "autorange_after_scroll" feature works, how does it express the visible min and max of the x axis? I just need exactly this feature, but while ignoring shapes.

I asked on the plotly forum but so far no one knew how to do this. They know how to define the range based on min and max value of a specific series of value, but no one could tell me so far how to "read" the currently visible min and max x...

**Reply by dbuezas** [reply]
It works in a very lazy way, it just removes all data beyond the visible range and then it tells plotly yo rescale.
To read the visible min/max, you need to grab the visible_range using "get",  filter out all data outside of it, and then compute the min/max. The issue is that if you use that to set the yaxis range, then it won't let you zoom. Plotly is rather limited in this regard, I am not sure your requirement can be accomplished.
Have you tried asking ChatGPT how to do this in plain plotly?
